src/components/DataBar/DataBar.tsx

```diff
@@ -0,0 +1,197 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './DataBar.module.css';
+
+import DataBarSegment from '../DataBarSegment';
+import type { Variants } from '../DataBarSegment';
+
+type DataBarSegmentProps = {
+  /**
+   * Tooltip text to be displayed when the segment is hovered.
+   */
+  text: string;
+  /**
+   * The value of the individual segment that accumulates towards the task goal.
+   */
+  value: number;
+  /**
+   * Color variant of the individual segment.
+   */
+  variant?: Variants;
+};
+
+export type Props = {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Max value to be represented by the data bar.
+   */
+  max: number;
+  /**
+   * A list of segments to be represented by the data bar.
+   */
+  segments: DataBarSegmentProps[];
+  /**
+   * Color variant of the data bar. Decorates the segments.
+   */
+  variant?: Variants;
+} & React.HTMLAttributes<HTMLElement>;
+
+/**
+ * ```ts
+ * import {DataBar} from "@chanzuckerberg/eds";
+ * ```
+ *
+ * A data bar component that represents the relative completion of a task represented by one or more segments.
+ *
+ * Example usage:
+ *
+ * ```tsx
+ * <DataBar
+ *  id="data-bar"
+ *  max={100}
+ *  segments={[
+ *    { value: 25, text: 'Segment 1' },
+ *    { value: 10, text: 'Segment 2' },
+ *    { value: 15, text: 'Segment 3' },
+ *  ]} />
+ * ```
+ */
+export const DataBar = ({
+  className,
+  max = 100,
+  segments,
+  variant = 'brand',
+  ...other
+}: Props) => {
+  /**
+   * Warns the developer if the data bar has no accessible name.
+   */
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    !other['id'] &&
+    !other['aria-label'] &&
+    !other['aria-labelledby']
+  ) {
+    console.warn('You must provide an accessible name for the data bar');
+  }
+
+  /**
+   * Calculates the total of the segment values.
+   */
+  const totalSegmentValue = segments.reduce(
+    (previousValue, segment) => previousValue + segment.value,
+    0,
+  );
+
+  const isFull = totalSegmentValue >= max;
+
+  /**
+   * Calculates the total width the segments will take up where the width is at least 5% of the max value.
+   */
+  let totalSegmentWidth = 0;
```

Comment:
looks like this is unused 👀  can this block be removed since you added the `segmentComponents` logic later?

Reply:
Dang think you're right, will remove